add property controller, dao, and integration tests

adamkorynta opened the pull request:
Adds features to handle properties in the Content Data API (CDA). It adds a new PropertyController and PropertyDao to manage access and manipulation of properties in the database layer.

Java client: https://github.com/HydrologicEngineeringCenter/cwms-data-api-client/pull/167

adamkorynta committed:
add property controller, dao, and integration tests

MikeNeilson commented on cwms-data-api/src/test/java/cwms/cda/api/PropertyControllerIT.java:
```diff
@@ -0,0 +1,258 @@
+                .contentType(Formats.JSONV2)
+                .body(json)
+                .header(AUTH_HEADER, user.toHeaderValue())
+                .when()
```
unindent .when and .then. (also looks like it's all indented twice.)

adamkorynta replied:
oops, missed one, updated

MikeNeilson reviewed (changes requested):
Some formatting got away from you, but otherwise looks great.

MikeNeilson commented on cwms-data-api/src/main/java/cwms/cda/api/PropertyController.java:
```diff
@@ -0,0 +1,296 @@
+
+    private static ObjectMapper getObjectMapperForFormat(ContentType contentType) {
+        ObjectMapper om;
+        if (ContentType.equivalent(Formats.JSONV2, contentType.toString())) {
```
The content-type mapper was design so you can just pass the data type in and with an annotation on the datatype get the right formatter given the content-type string.

This function is duplication of all that logic.

adamkorynta replied:
Looks like that was only done one-way. I implemented the deserialization and updated my code to use the Formats factory methods. I'll do a follow-up PR removing the redundant code that I had copied this logic from.

adamkorynta committed:
implement parser logic to follow formatter factory logic

this simplifies how parsing and formatting is done so that we can remove duplicate code

MikeNeilson reviewed (approved):
Nice, thanks for the additional improvement for parsing the content as well.
